Corona-Warn-App/src/main/java/de/rki/coronawarnapp/vaccination/ui/scan/VaccinationQrCodeScanFragment.kt

```diff
@@ -0,0 +1,161 @@
+package de.rki.coronawarnapp.vaccination.ui.scan
+
+import android.Manifest
+import android.content.pm.PackageManager
+import android.os.Bundle
+import android.view.View
+import android.view.accessibility.AccessibilityEvent.TYPE_ANNOUNCEMENT
+import androidx.core.view.isGone
+import androidx.fragment.app.Fragment
+import com.google.zxing.BarcodeFormat
+import com.journeyapps.barcodescanner.DefaultDecoderFactory
+import de.rki.coronawarnapp.R
+import de.rki.coronawarnapp.bugreporting.ui.toErrorDialogBuilder
+import de.rki.coronawarnapp.databinding.FragmentScanQrCodeBinding
+import de.rki.coronawarnapp.util.DialogHelper
+import de.rki.coronawarnapp.util.di.AutoInject
+import de.rki.coronawarnapp.util.permission.CameraPermissionHelper
+import de.rki.coronawarnapp.util.ui.doNavigate
+import de.rki.coronawarnapp.util.ui.popBackStack
+import de.rki.coronawarnapp.util.ui.viewBindingLazy
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModelFactoryProvider
+import de.rki.coronawarnapp.util.viewmodel.cwaViewModels
+import javax.inject.Inject
+
+class VaccinationQrCodeScanFragment :
+    Fragment(R.layout.fragment_scan_qr_code),
+    AutoInject {
+
+    @Inject lateinit var viewModelFactory: CWAViewModelFactoryProvider.Factory
+    private val viewModel: VaccinationQrCodeScanViewModel by cwaViewModels { viewModelFactory }
+
+    private val binding: FragmentScanQrCodeBinding by viewBindingLazy()
+    private var showsPermissionDialog = false
+
+    override fun onViewCreated(
+        view: View,
+        savedInstanceState: Bundle?
+    ) {
+        with(binding) {
+            qrCodeScanTorch.setOnCheckedChangeListener { _, isChecked ->
+                binding.qrCodeScanPreview.setTorch(isChecked)
+            }
+
+            qrCodeScanToolbar.setNavigationOnClickListener { popBackStack() }
+            qrCodeScanPreview.decoderFactory = DefaultDecoderFactory(listOf(BarcodeFormat.QR_CODE))
+            qrCodeScanViewfinderView.setCameraPreview(binding.qrCodeScanPreview)
+        }
+
+        viewModel.event.observe(this) { event ->
```

Comment:
`viewlifecycleowner` please ? There should be a Lint rule to avoid that already ,not sure why it is not working 🤔

Reply:
but is not `observe2 ` creates an extra observer object ?